File: test/Aqua.Tests/TypeSystem/Extensions/Using_type_extensions.cs
// Copyright (c) Christof Senn. All rights reserved. See license.txt in the project root for license information.

namespace Aqua.Tests.TypeSystem.Extensions
{
    using Aqua.TypeSystem.Extensions;
    using Shouldly;
    using Xunit;

    public class Using_type_extensions
    {
        private class CustomType
        {
        }

        [Fact]
        public void Is_anonymous_should_return_true_for_anonymous_type()
        {
            new { X = 1 }.GetType().IsAnonymousType().ShouldBeTrue();
        }

        [Fact]
        public void Is_anonymous_should_return_false_for_custom_type()
        {
            new CustomType().GetType().IsAnonymousType().ShouldBeFalse();
        }

        [Fact]
        public void Is_anonymous_should_return_false_for_string_type()
        {
            typeof(string).IsAnonymousType().ShouldBeFalse();
        }
    }
}

File: test/Aqua.Tests.VBNet/TypeSystem/Extensions/Using_type_extensions.vb
Imports Aqua.TypeSystem.Extensions
Imports Shouldly
Imports Xunit

Namespace Aqua.Test.TypeSystem.Extensions
    Public Class Using_type_extensions
        Private Class CustomType
        End Class

        <Fact>
        Public Sub Is_anonymous_should_return_true_for_anonymous_type()
            Dim obj = New With {Key .X = 0}
            obj.GetType().IsAnonymousType().ShouldBeTrue()
        End Sub

        <Fact>
        Public Sub Is_anonymous_should_return_false_for_custom_type()
            Dim obj = New CustomType
            obj.GetType().IsAnonymousType().ShouldBeFalse()
        End Sub

        <Fact>
        Public Sub Is_anonymous_should_return_false_for_string_type()
            Dim obj = ""
            obj.GetType().IsAnonymousType().ShouldBeFalse()
        End Sub
    End Class
End Namespace
